### PR TITLE
Feature/testing

### PR DESCRIPTION
**Pull request by MasumaJaffery**
Hello Code Reviewer!
I hope you are doing well!
In this Milestone;
1. Install bundle install
2. Write Tests for Associations and Methods(Callbacks).
3.  Run Test cases by bundle exec rspec in wsl ubuntu terminal.
Thanks for your Time to review my PR!
Happy Reviewing! 🤓 
Stay Blessed. ⭐

### CODE REVIEW

File: app/models/user.rb
Comment (aamirkhan2478) [inline app/models/user.rb:1]: https://github.com/MasumaJaffery/Blog_App/blob/8b36b38c7ebd21264b02d31996aa3e242593e683/app/models/post.rb#L1-L13

- [x] Please add validation to post model as well for now you missed validation of post model so kindly fix it. The following are the post model validations:

- `Post` model has validations:
  - `Title` must not be `blank`.
  - `Title` must not exceed `250` characters.
  - `CommentsCounter` must be an integer `greater than` or `equal` to `zero`.
  - `LikesCounter` must be an integer `greater than` or `equal` to `zero`.

File: spec/post_spec.rb
Comment (aamirkhan2478) [inline spec/post_spec.rb:1]: - [x] Kindly write unit tests for validations as well for now you didn't write any validation tests for the post model so kindly fix this issue as well :slightly_smiling_face:

**Review by aamirkhan2478 — CHANGES_REQUESTED** [review]
Hi @MasumaJaffery 



Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Good Effort

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @MasumaJaffery, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] All requested changes are made ✔️ 
- [ ] No linters errors ✔️ 
- [ ] All tests are passing ✔️ 
- [ ] Correct Gitflow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**